discord/message.py

```diff
@@ -1760,20 +1844,20 @@ class PartialMessage(Hashable):
     __slots__ = ('channel', 'id', '_cs_guild', '_state')
 
     jump_url: str = Message.jump_url  # type: ignore
-    edit = Message.edit
-    add_files = Message.add_files
-    remove_attachments = Message.remove_attachments
-    delete = Message.delete
-    publish = Message.publish
-    pin = Message.pin
-    unpin = Message.unpin
-    add_reaction = Message.add_reaction
-    remove_reaction = Message.remove_reaction
-    clear_reaction = Message.clear_reaction
-    clear_reactions = Message.clear_reactions
-    reply = Message.reply
-    to_reference = Message.to_reference
-    to_message_reference_dict = Message.to_message_reference_dict
+    edit: _EditCallback = Message.edit  # type: ignore
+    add_files: _AddFilesCallback = Message.add_files  # type: ignore
+    remove_attachments: _RemoveAttachmentsCallback = Message.remove_attachments  # type: ignore
+    delete: _DeleteCallback = Message.delete  # type: ignore
+    publish: Callable[[], Coroutine[Any, Any, None]] = Message.publish  # type: ignore
+    pin: _PinCallback = Message.pin  # type: ignore
+    unpin: _UnpinCallback = Message.unpin  # type: ignore
+    add_reaction: _AddReactionCallback = Message.add_reaction  # type: ignore
+    remove_reaction: _RemoveReactionCallback = Message.remove_reaction  # type: ignore
+    clear_reaction: _ClearReactionCallback = Message.clear_reaction  # type: ignore
+    clear_reactions: Callable[[], Coroutine[Any, Any, None]] = Message.clear_reactions  # type: ignore
+    reply: _ReplyCallback = Message.reply  # type: ignore
+    to_reference: _ToReferenceCallback = Message.to_reference  # type: ignore
+    to_message_reference_dict: Callable[[], MessageReferencePayload] = Message.to_message_reference_dict  # type: ignore
```

Comment:
I am not a fan of this change.

Reply:
The type ignores can be removed using typing.Self fyi

Reply:
Would it be better to type these to `Any` instead?

Reply:
I reverted this change.